Dockerfile

```diff
@@ -1,30 +1,63 @@
-# Build the environment.
-FROM node:20-alpine as production
+# Install dependencies only when needed
+FROM node:20-alpine AS builder
+RUN apk add --no-cache libc6-compat
 
-# Install git to resolve issues installing the
-# nypl/dgx-header-component package.
-RUN apk add git
+WORKDIR /app
+# Create the directory as root
+RUN mkdir -p ./log 
 
-WORKDIR /
-
-# Set environment variables. NODE_ENV is set early because we
-# want to use it when running `npm install` and `npm run build`.
-ENV PATH /app/node_modules/.bin:$PATH
-ENV PORT=3000 \
-    NODE_ENV=production
-
-# Install dependencies.
 COPY package.json ./
 COPY package-lock.json ./
-RUN npm ci --legacy-peer-deps
 
-# Copy the app files.
-COPY . ./
+RUN npm ci --cache .npm --legacy-peer-deps
 
-EXPOSE $PORT
+# Copy the source code
+COPY . .
+
+# expose the git SHA and branch as env vars
+# we use the same names that vercel uses for cross-compatibility in deployment
+ARG NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA
+ARG NEXT_PUBLIC_CM_URL
+ARG NEXT_PUBLIC_ADOBE_ANALYTICS_URL
+ARG SENDGRID_API_KEY
+ARG NEW_RELIC_API_KEY
+ARG NEW_RELIC_APP_NAME
+ARG NEW_RELIC_LICENSE_KEY
+ARG RELEASE_STAGE
 
-# Build the app!
 RUN npm run build
 
-# CMD is the default command when running the docker container.
-CMD npm start
+# RUNNER
+FROM node:20-alpine AS runner
+
+
+# Disable Next.js telemetry during runtime.
+ENV NEXT_TELEMETRY_DISABLED 1
+
+# We need to expose these ARG again, they are needed for browser monitor
+ARG NEW_RELIC_APP_NAME
+ARG NEW_RELIC_LICENSE_KEY
+ENV NEW_RELIC_APP_NAME $NEW_RELIC_APP_NAME
+ENV NEW_RELIC_LICENSE_KEY $NEW_RELIC_LICENSE_KEY
+
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+
+COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
+COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static
+COPY --from=builder --chown=nextjs:nodejs /app/public ./public
```

Comment:
For my understanding, do you know if we need to copy over `node_modules`, or if we don't, why we do not? If this is working then based on my limited understanding, I would assume the required dependencies are already built into compiled js in public and static files?